includes/functions/functions_search.php

```diff
@@ -140,77 +146,80 @@ function zen_parse_search_string($search_str = '', &$objects = array()) {
     $keyword_count = 0;
     $operator_count = 0;
     $balance = 0;
-    for($i=0; $i<count($objects); $i++) {
-        if ($objects[$i] == '(') $balance --;
-        if ($objects[$i] == ')') $balance ++;
-        if ( ($objects[$i] == 'and') || ($objects[$i] == 'or') ) {
-            $operator_count ++;
-        } elseif ( (is_string($objects[$i]) && $objects[$i] == '0') || ($objects[$i]) && ($objects[$i] != '(') && ($objects[$i] != ')') ) {
-            $keyword_count ++;
+    for ($i = 0; $i < $j + 1; $i++) {
```

Comment:
Should have been:
````php
   for ($i = 0; $i < $j + 2; $i++) {
````
Because: Up above, `$j` is equal to the count of `$objects` minus one. But then `$objects` gets another member. So where previously this line forced a count of `$objects` to "reset" `$j`, `$j` is just 2 too small at this point. Missing one from the previous subtraction and then missing one from the additional member. 

I mean all in the sake of micro-optimization right?